cylc/flow/network/schema.py

```diff
@@ -1598,81 +1680,244 @@ async def gen(*args: Any, **kwargs: Any) -> AsyncGenerator[Any, None]:
             AsyncGenerator[Any, None]: an async generator that will
                 yield values from resolvers.
         """
-        while True:
-            yield await func(*args, **kwargs)
-            await asyncio.sleep(sleep_seconds)
+        try:
+            while True:
+                yield await func(*args, **kwargs)
+                await asyncio.sleep(sleep_seconds)
+        finally:
+            yield None
     return gen
 
 
+def delta_subs(root, info, **args) -> AsyncGenerator[Any, None]:
+    return info.context.get('resolvers').subscribe_delta(info, args)
+
+
+class Pruned(ObjectType):
+    class Meta:
+        description = """WFS Nodes/Edges that have been removed."""
+    families = List(String, default_value=[])
+    family_proxies = List(String, default_value=[])
+    jobs = List(String, default_value=[])
+    tasks = List(String, default_value=[])
+    task_proxies = List(String, default_value=[])
+    edges = List(String, default_value=[])
+
+
+class Added(ObjectType):
+    class Meta:
+        description = """Added node/edge deltas."""
+
+    families = List(
+        Family,
+        description="""Family definitions.""",
+        args=DEF_ARGS,
+        delta_type=String(default_value=DELTA_ADDED),
```

Comment:
The `delta_type` appears to be the only difference between something added or updated?

I was thinking in using fragments to define the families, tasks, jobs, etc, in my subscription/query. But then found out that I couldn't, because I would end up with

```
fragment CyclePoint1 on Updated {
  cyclePoint: familyProxies (ids: ["root"]) {
    cyclePoint
  }
}

fragment CyclePoint2 on Added {
  cyclePoint: familyProxies (ids: ["root"]) {
    cyclePoint
  }
}
```

As there is no common type, I cannot use a single fragment for both. Looking here, I think the `delta_type` is the only difference between entities that were added or updated?

Is it something that would make sense to unify @dwsutherland ? Although I think if we ever had a case that updated was different than added, then it would be a mess to undo that right... probably better to have a bit of duplication in the UI

Reply:
We can just create a Union of types like I did with the graph nodes (which was a Union of `FamilyProxy` and `TaskProxy`).

Reply:
> Is it something that would make sense to unify @dwsutherland ? Although I think if we ever had a case that updated was different than added, then it would be a mess to undo that right... probably better to have a bit of duplication in the UI

I think the whole point in separating deltas into updated and added was because they were different, and we wanted the ability to subscribe to different fields and structure in both cases or perhaps the client would have different actions for both...

I don't think you can have an added & updated in one delta for the same element.. but yes Union of types might do it I'd have to see..